pydicom/jsonrep.py

```diff
@@ -41,3 +47,202 @@ def convert_to_python_number(value, vr):
         else:
             value = number_type(value)
     return value
+
+
+class JsonDataElementConverter(object):
+    """Handles conversion between JSON struct and :class:`DataElement`."""
+
+    def __init__(self, dataset_class, tag, vr, value, value_key,
+                 bulk_data_uri_handler):
+        """Create a new converter instance.
+
+        Parameters
+        ----------
+        dataset_class : Dataset derived class
+            Class used to create sequence items.
+        tag : BaseTag
+            The data element tag.
+        vr : str
+            The data element value representation.
+        value : list
+            The data element's value(s).
+        value_key : str or None
+            Key of the data element that contains the value
+            (options: ``{"Value", "InlineBinary", "BulkDataURI"}``)
+        bulk_data_uri_handler: callable or None
+            Callable function that accepts the "BulkDataURI" of the JSON
+            representation of a data element and returns the actual value of
+            that data element (retrieved via DICOMweb WADO-RS)
+        """
+        self.dataset_class = dataset_class
+        self.tag = tag
+        self.vr = vr
+        self.value = value
+        self.value_key = value_key
+        self.bulk_data_uri_handler = bulk_data_uri_handler
+
+    def get_element_values(self):
+        """Return a the data element value or list of values.
+
+        Returns
+        -------
+        str or bytes or int or float or dataset_class
+        or PersonName3 or PersonNameUnicode or list of any of these types
+            The value or value list of the newly created data element.
+        """
+        if self.value_key == 'Value':
+            if not isinstance(self.value, list):
+                fmt = '"{}" of data element "{}" must be a list.'
+                raise TypeError(fmt.format(self.value_key, self.tag))
+            if not self.value:
+                return empty_value_for_VR(self.vr)
+            element_value = []
+            for v in self.value:
+                element_value.append(self.get_regular_element_value(v))
```

Comment:
`element_value = [self.get_regular_element_value(v) for v in self.value]`

Reply:
Of course.